spec/services/post_alerter_spec.rb

```diff
@@ -1871,6 +1871,72 @@ def set_topic_notification_level(user, topic, level_name)
       end
     end
 
+    context "with category and tags" do
+      fab!(:muted_category) do
+        Fabricate(:category).tap do |category|
+          CategoryUser.set_notification_level_for_category(
+            user,
+            CategoryUser.notification_levels[:muted],
+            category.id,
+          )
+        end
+      end
+      fab!(:topic_with_muted_tag_and_watched_category) { Fabricate(:topic, category: category) }
```

Comment:
I think the fabrication here can be improved as well. It'll be better to fabricate the tags first and then just create the topic with the muted tag instead of manually creating the `TopicTag` record below when fabricating the tag.